### PR TITLE
Add autoFormat option to PhoneInput

### PR DESCRIPTION
**Pull request by jling90**
Adds an 'autoFormat' option to PhoneInput which allows phone numbers to be formatted as they are entered.

#
|Before|After|
|-------|-----|
|before|after|

### CODE REVIEW

**Comment by kelvinoshea** [comment]
this is awesome

**Comment by bro-d** [comment]
i never thought i would see the day.

**Comment by sdcoffey** [comment]
This is awesome! Would love to see this merged!

**Comment by MacKentoch** [comment]
Seems nice, please ship it 🙏

**Comment by brunitob** [comment]
Any news?

**Review by cKurultayKalkan — APPROVED** [review]

**Comment by madhusvce07** [comment]
Any news when would it be merged so as to use it??

**Comment by jling90** [comment]
image

gif

**Comment by brunitob** [comment]
Image of bob

**Comment by michaelxtendly** [comment]
PLEASE we need this feature :)

**Comment by michaelVictoriaDev** [comment]
@thegamenicorus  please Accept the PR :)

**Comment by gucolin** [comment]
I ended up just pasting the fix into my component, make sure you do `clear()` at the end though.

``` javascript
  onCountrySelected = (country) => {
    this.phoneInput.selectCountry(country.cca2.toLowerCase());
    this.phoneNumberFormatter = new asYouTypeFormatter(country.cca2);
    this.setState({
      cca2: country.cca2
    });
  }

  onPhoneNumberChanged = (phoneNumber) => {
    let formattedNumber;

    phoneNumber.replace(/-/g, '')
      .replace(/ /g, '')
      .split('')
      .forEach(n => formatted = this.phoneNumberFormatter.inputDigit(n));

    this.setState({
      phoneNumber: formattedNumber
    });

    this.phoneNumberFormatter.clear();
  }
```

**Comment by rafaismyname** [comment]
@thegamenicorus are you alive?

File: lib/phoneNumber.js [hunk 1]
```diff
@@ -86,6 +88,18 @@ class PhoneNumber {
...
 
+  format(number, iso2) {
+    const formatter = new asYouTypeFormatter(iso2)
+    let formatted;
+
+    number.replace(/-/g, '')
+      .replace(/ /g, '')
+      .split('')
+      .forEach(n => formatted = formatter.inputDigit(n));
+
```
Comment (mths0x5f) [inline lib/phoneNumber.js]: is not the case of this advice?

https://github.com/thegamenicorus/react-native-phone-input/pull/70#issuecomment-457857601

**Comment by michaelVictoriaDev** [comment]
@thegamenicorus THANK YOU!!!!!!

**Comment by Zakyyy** [comment]
is it merged? @thegamenicorus

**Pull request merged**